tensorboard/plugins/graph/graphs_plugin_test.py

```diff
@@ -198,41 +239,75 @@ def test_graph_large_attrs(self):
     self.assertEqual({'message_prefix': [b'value']},
                      large_attrs)
 
-  def test_run_metadata(self):
-    self.set_up_with_runs()
-    (metadata_pbtxt, mime_type) = self.plugin.run_metadata_impl(
+  @with_runs()
+  def test_run_metadata(self, plugin):
+    (metadata_pbtxt, mime_type) = plugin.run_metadata_impl(
         self._RUN_WITH_GRAPH, self._METADATA_TAG)
     self.assertEqual(mime_type, 'text/x-protobuf')
     text_format.Parse(metadata_pbtxt, config_pb2.RunMetadata())
     # If it parses, we're happy.
 
-  def test_is_active_with_graph_without_run_metadata(self):
-    self.generate_run('w_graph_wo_meta',
+  @with_runs()
+  def test_is_active_with_graph_without_run_metadata(self, plugin):
+    logdir = self.get_temp_dir()
+    self.generate_run(logdir,
+                      'w_graph_wo_meta',
                       include_graph=True,
                       include_run_metadata=False)
```

Comment:
I don’t see how this is read. From what I can tell, the flow is that the
`with_runs()` decorator calls `set_up_with_runs` with no arguments,
which calls `generate_run` with default arguments and then uses
`bootstrap_plugin` to create and return a multiplexer. This ensures that
the multiplexer has read the relevant data, because `bootstrap_plugin`
calls `Reload`, which is blocking. But these `generate_runs` aren’t
followed by a `Reload` on the multiplexer used by the plugin (either
directly through its context or encapsulated in a data provider). How is
this data getting to the plugin?

Should it maybe be the case that the `with_runs` decorator factory takes
the same kwargs as `set_up_with_runs`, and takes care of generating only
the specified run (`with_run`, then, I guess?)? This would be more in
line with its use in other plugins.

Reply:
Oops, these tests were completely broken.  I had gotten the `test_is_active_without_both` case working in an ugly way but hadn't done the rest of these parallel cases (they were passing, but for the wrong reason obviously).

Anyway I refactored this substantially; PTAL.